gp2gp-translator/src/test/java/uk/nhs/adaptors/pss/translator/mapper/ConditionMapperTest.java

```diff
@@ -0,0 +1,230 @@
+    private static final String OBSERVATION_ID = "9f25604b-d2e8-402a-aae2-c78879c3e647";
+    private static final String LINKSET_ID = "DCC26FC9-4D1C-11E3-A2DD-010000000161";
+    private static final String CODING_DISPLAY = "THIS IS A TEST";
+    private static final String PRACTISE_CODE = "TEST123";
+    private static final Date EHR_EXTRACT_AVAILABILITY = new Date(2022, Calendar.JANUARY, 1);
+    private static final DateTimeType EHR_EXTRACT_AVAILABILITY_DATETIME = new DateTimeType(EHR_EXTRACT_AVAILABILITY);
+
+    private static final int RELATED_CLINICAL_CONTENT_COUNT = 3;
+    private static final int NOTES_COUNT = 4;
+
+    private static final String ACTUAL_PROBLEM_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ActualProblem-1";
+    private static final String PROBLEM_SIGNIFICANCE_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-ProblemSignificance-1";
+    private static final String RELATED_CLINICAL_CONTENT_URL = "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-RelatedClinicalContent-1";
+
+    @Mock
+    private CodeableConceptMapper codeableConceptMapper;
+    @Mock
+    private DateTimeMapper dateTimeMapper;
+
+    @InjectMocks
+    private ConditionMapper conditionMapper;
+
+    @BeforeEach
+    public void setUp() {
+        var codeableConcept = new CodeableConcept();
+        var coding = new Coding();
+        coding.setDisplay(CODING_DISPLAY);
+        codeableConcept.addCoding(coding);
+
+        when(codeableConceptMapper.mapToCodeableConcept(any())).thenReturn(codeableConcept);
+    }
+
+    @Test
+    public void When_MappingAllValidData_Expect_AppropriateOutput() {
+        when(dateTimeMapper.mapDateTime(any())).thenReturn(EHR_EXTRACT_AVAILABILITY_DATETIME);
+        var linkset = unmarshallLinkset("linkset_valid.xml");
+        var observationStatement = unmarshallObservationStatement("observationStatement_1.xml");
+        var ehrComposition = new RCMRMT030101UK04EhrComposition();
+        var component = new RCMRMT030101UK04Component4();
+
+        component.setLinkSet(linkset);
+        ehrComposition.setComponent(List.of(component));
+
+        var paramsBuilder = ConditionMapper.ConditionMapperParameters.builder();
+
+        paramsBuilder.ehrComposition(ehrComposition);
+        paramsBuilder.linkedObservationStatement(Optional.of(observationStatement));
+        paramsBuilder.ehrExtractAvailabilityTime(EHR_EXTRACT_AVAILABILITY);
+        paramsBuilder.actualProblem(Optional.of(createObservationWithId()));
+        paramsBuilder.relatedClinicalContent(List.of(
+            createObservationWithId(),
+            createImmunizationWithId(),
+            createReferralRequestWithId()
+        ));
+        paramsBuilder.patientId(PATIENT_ID);
+        paramsBuilder.encounterId(Optional.of(ENCOUNTER_ID));
+        paramsBuilder.asserterId(ASSERTER_ID);
+        paramsBuilder.practiseCode(PRACTISE_CODE);
```

Comment:
Why not chain all of it?
```suggestion
        var paramsBuilder = ConditionMapper.ConditionMapperParameters.builder()
            .ehrComposition(ehrComposition)
            .linkedObservationStatement(Optional.of(observationStatement))
            ....
            .build();
```